onnxruntime/hosting/include/beast_http.h

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <functional>
+#include <iostream>
+#include <memory>
+#include <string>
+#include <thread>
+#include <vector>
+
+#include "util.h"
+#include "http_context.h"
+#include "routes.h"
+#include "http_session.h"
+#include "listener.h"
+
+namespace http = beast::http;      // from <boost/beast/http.hpp>
+namespace net = boost::asio;       // from <boost/asio.hpp>
+using tcp = boost::asio::ip::tcp;  // from <boost/asio/ip/tcp.hpp>
+using handler_fn = std::function<void(std::string, std::string, std::string, Http_Context&)>;
+
+// Accepts incoming connections and launches the sessions
+class App {
+ public:
+  App() {
+    address_ = boost::asio::ip::make_address_v4("0.0.0.0");
+    port_ = 8080;
+    threads_ = std::thread::hardware_concurrency();
+  }
+
+  App& bind(net::ip::address address, unsigned short port) {
+    address_ = std::move(address);
+    port_ = port;
+    return *this;
+  }
+
+  App& num_threads(int threads) {
+    threads_ = threads;
+    return *this;
+  }
+
+  App& post(const std::regex& route, handler_fn fn) {
+    //    routes->http_posts[route] = std::move(fn);
+    routes->register_controller(http::verb::post, route, fn);
+    return *this;
+  }
+
+  App& run() {
+    net::io_context ioc{threads_};
+    // Create and launch a listening port
+    std::make_shared<listener>(routes, ioc, tcp::endpoint{address_, port_})->run();
+
+    std::cout << "Listening at: \n"
```

Comment:
Need to be updated to use logger. TODO is fine here for this PR.

Reply:
Added a TODO